storage/local/storage.go

```diff
@@ -549,7 +556,14 @@ func (s *memorySeriesStorage) Append(sample *clientmodel.Sample) {
 		s.fpLocker.Unlock(rawFP)
 		s.fpLocker.Lock(fp)
 	}
-	series := s.getOrCreateSeries(fp, sample.Metric)
+	series, lastTime := s.getOrCreateSeries(fp, sample.Metric)
+
+	if sample.Timestamp <= lastTime {
+		s.fpLocker.Unlock(fp)
+		log.Warnf("Ignoring sample with out-of-order timestamp for fingerprint %v: %v is not after %v", fp, sample.Timestamp, lastTime)
```

Comment:
It's helpful to log the series, too, as the fp alone is not very helpful for debugging. ("Where did this out-of-order sample come from?")